clients/js/src/index.ts

```diff
@@ -39,3 +39,61 @@ export class CacReader {
         return deepMerge(targetConfig, ...requiredOverrides);
     }
 }
+
+export class ExperimentReader {
+    experiments: ExperimentStore;
+
+    constructor(experiments: Experiments) {
+        this.experiments = new Map(experiments.map(exp => [exp.id, exp]));
+    }
+
+    public getApplicableVariant(data: IObject, toss: number): Array<String> {
+        const experiments = this.getSatisfiedExperiments(data);
+        const variants = [];
+        for (const exp of experiments) {
+            const v = this.decideVariant(exp.traffic_percentage, exp.variants, toss);
+            if (v) {
+                variants.push(v.id)
+            }
+        }
+        return variants;
+    };
+
+    getSatisfiedExperiments(data: IObject): Experiments {
+        const experiments = [];
+        const iter = this.experiments.values();
+
+        while (true) {
```

Comment:
Can we just use a for loop? Infinite loops are dangerous and there are a limited number of experiments